### PR TITLE
[MIN-72] Create function to calculate readability score

### PR DESCRIPTION
**Pull request by osw282**
This PR kick-start the development of our metric service by integrating a feature to compute the readability of a piece of text. It uses the [Flesch-Kincaid readability](https://en.wikipedia.org/wiki/Flesch%E2%80%93Kincaid_readability_tests) score with the help of the [textstat](https://github.com/textstat/textstat) module.

What's new:
- We've added a new `monitoring` folder at the project root, housing the `metric_service.py` script.
- Within `metric_service.py,` there's a new function named `compute_readability()`. This function takes a single argument - the text for which readability needs to be analysed, and it returns a score ranging from 1 to 100.
- A corresponding test has been implemented for the `compute_readability()` function.

### CODE REVIEW

File: monitoring/metric_service.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        llm_response (str): the model's response post by Streamlit
+
+    Raises:
+        TypeError: raise if the model response received is not a string
+        ValueError: raise if the model response received is a empty string
+
+    Returns:
+        float: the readability score of the response
+
+    Meaning of scores:
```
Comment (JonoCX) [inline monitoring/metric_service.py]: This should probably be above "Args" rather than down here.

File: tests/test_monitoring/test_metric_service.py
Comment (JonoCX) [inline tests/test_monitoring/test_metric_service.py:1]: There should probably be a test to for a good and bad score

**Review by JonoCX — CHANGES_REQUESTED** [review]
Nice - see comments

**Review by JonoCX — APPROVED** [review]
Nice - LGTM

**Review by dudeperf3ct — APPROVED** [review]
LGTM! Nice work @osw282

**Pull request merged**